### ISSUE TITLE
FrenchRevolutionaryCalendarVonMadlerTest failures

### ISSUE THREAD

**Issue by PyvesB** [caarmen/french-revolutionary-calendar#17]
Hello there!

I've just tried to build the project but there seem to be some failures with the Von Madler tests:
```
[ERROR] Tests run: 30, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.118 s <<< FAILURE! - in ca.rmen.lfrc.FrenchRevolutionaryCalendarVonMadlerTest
[ERROR] testFrenchDate19700923(ca.rmen.lfrc.FrenchRevolutionaryCalendarVonMadlerTest)  Time elapsed: 0.008 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<17[9-01-01]> but was:<17[8-13-05]>
	at ca.rmen.lfrc.FrenchRevolutionaryCalendarVonMadlerTest.testFrenchDate19700923(FrenchRevolutionaryCalendarVonMadlerTest.java:286)

[ERROR] testNewYears(ca.rmen.lfrc.FrenchRevolutionaryCalendarVonMadlerTest)  Time elapsed: 0.016 s  <<< FAILURE!
java.lang.AssertionError: 
Expected 177-03-01 but got 177-2-30
Expected 177-05-01 but got 177-4-30
Expected 178-01-02 but got 178-1-1
Expected 178-01-09 but got 178-1-8
Expected 178-01-01 but got 177-13-5
Expected 177-13-5, but got 177-13-4
Expected 178-03-01 but got 178-2-30
Expected 178-05-01 but got 178-4-30
Expected 179-01-02 but got 179-1-1
Expected 179-01-09 but got 179-1-8
Expected 179-01-01 but got 178-13-5
Expected 178-13-5, but got 178-13-4
Expected 179-03-01 but got 179-2-30
Expected 179-05-01 but got 179-4-30
Expected 180-01-02 but got 180-1-1
Expected 180-01-09 but got 180-1-8
Expected 180-01-01 but got 179-13-5
Expected 179-13-5, but got 179-13-4

	at ca.rmen.lfrc.FrenchRevolutionaryCalendarVonMadlerTest.testNewYears(FrenchRevolutionaryCalendarVonMadlerTest.java:136)
```

Am I missing something?

Thanks in advance,

Pyves

**Comment by caarmen** [comment]
I remember seeing inconsistent test results between different versions of the jdk. Time zone rules change, and are reflected in releases of the jdk.

What version of the jdk are you using? The tests pass for me (`mvn clean test`) using the following:
* 1.8.0_112
* 1.8.0_141
* 1.8.0_152
* 1.8.0_161

For jdks 9.0.4 and 10, it won't compile. That requires a kotlin upgrade (`library/pom.xml`), and also changing the min java version from 1.5 to 1.6 (`pom.xml`). I'm not sure I want to drop 1.5 support though... However, if I do make these changes locally, the tests pass.

**Comment by caarmen** [comment]
Actually, if I put my computer's time zone to London, instead of Paris, I get the von madler test failures :)

**Comment by PyvesB** [comment]
I'm using JDK 1.8.0_162. Indeed my computer's timezone is set to London, this must be the cause. 😉

**Comment by caarmen** [comment]
I've done a fix in branch issue-17.  I tested running `mvn clean test` in multiple time zones. Seems ok now, at least in the time zones I tested :)

**Comment by PyvesB** [comment]
All tests are now passing for me, good job!

**Comment by caarmen** [comment]
Will be included in release 1.8.1.
